<!-- Panda Autograsp default launchfile-->

<launch>

    <!-- Include sensor package and grasp detection package launch files-->
    <include file="$(find panda_autograsp)/launch/kinect2_bridge.launch" >
        <arg name="output" value="log" />
    </include>

    <!-- Load panda_autograsp package nodes -->

    <!-- Include rviz movit launchfile -->
    <!-- TODO: I still have to replace the parameters in this file to make it work on the real robot -->
    <!-- TODO: Change rviz_tutorials variable -->
    <include file="$(find panda_autograsp)/launch/panda_autograsp_rviz.launch">
        <arg name="output" value="log" />
    </include>

    <!-- Start Tf visual tool node-->
    <!-- <node pkg="tf_visual_tools" type="tf_visual_tools_publisher" name="rviz_tfs" output="screen" /> -->

    <!-- Create static transform for the calibration board-->
    <!-- TODO: Replace with math-->
    <!-- <arg name="alpha" value="${135/180*pi}" /> -->
    <!-- <arg name="beta" value="${135/180*pi}" /> -->
    <!-- <arg name="gamma" value="${135/180*pi}" /> -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="chessboard_frame_broadcaster" args="0.74 -0.10 0 0 0 0 panda_link0 chessboard_frame" />
    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="kinect2_rgb_optical_frame" args="0 1 0.82 0 0 2.3561899994543 panda_link0 kinect2_rgb_optical_frame" /> -->

    <!-- Grasp planner service server -->
    <!-- <node name="grasp_planner_server" pkg="panda_autograsp" type="grasp_planner_server.py" output="screen">
        <param name="grasp_detection_srv" value="$(arg grasp_detection_srv)" />
    </node> -->

    <!--Start panda_autograsp servers-->
    <node name="panda_autogras_server" pkg="panda_autograsp" type="panda_autograsp_server.py" output="screen"/>
    <node name="grasp_planner" pkg="panda_autograsp" type="grasp_planner_server.py" output="log"/>
    <node name="moveit_planner_server" pkg="panda_autograsp" type="moveit_planner_server.py" output="screen"/>

    <!-- Grasp planner service client -->
    <!-- <node name="grasp_planner_client" pkg="panda_autograsp" type="grasp_planner_client.py" output="screen" >
        <param name="grasp_img_quality" value="$(arg grasp_img_quality)" />
        <param name="grasp_detection_srv" value="$(arg grasp_detection_srv)" />
    </node> -->

</launch>